src/Silverback.Integration/Messaging/Diagnostics/ActivityEnricherFactory.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) 2020 Sergio Aquilini
+// This code is licensed under MIT license (see LICENSE file for details)
+
+using System;
+using System.Diagnostics;
+using Silverback.Messaging.Broker.Behaviors;
+
+namespace Silverback.Messaging.Diagnostics
+{
+    internal class ActivityEnricherFactory : IActivityEnricherFactory
+    {
+        private static readonly NullEnricher NullEnricherInstance = new();
+
+        private readonly IServiceProvider _serviceProvider;
+
+        public ActivityEnricherFactory(IServiceProvider serviceProvider)
+        {
+            _serviceProvider = serviceProvider;
+        }
+
+        public IBrokerActivityEnricher GetActivityEnricher(Type endpointType)
+        {
+            Type closedEnricherType = typeof(IBrokerActivityEnricher<>).MakeGenericType(endpointType);
```

Comment:
I think that this could/should be cached in a simple static `Dictionary<Type, Type>`. WDYT @mjeanrichard?

Reply:
Yeah, I was wondering about that as well. Do you have any idea of the performance of `MakeGenericType`?

Reply:
I did a quick benchmark and it seems a lot faster with a (concurrent) dictionary. I'll change it.